integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala

```diff
@@ -193,6 +196,30 @@ class CarbonIndexFileMergeTestCase
     sql("select * from mitable").show()
   }
 
+  // CARBONDATA-2704, test the index file size after merge
+  test("Verify the size of the index file after merge") {
+    sql("DROP TABLE IF EXISTS fileSize")
+    sql(
+      """
+        | CREATE TABLE fileSize(id INT, name STRING, city STRING, age INT)
+        | STORED BY 'org.apache.carbondata.format'
+        | TBLPROPERTIES('SORT_COLUMNS'='city,name')
+      """.stripMargin)
+    sql(s"LOAD DATA LOCAL INPATH '$file2' INTO TABLE fileSize OPTIONS('header'='false')")
+    val table = CarbonMetadata.getInstance().getCarbonTable("default", "fileSize")
+    var loadMetadataDetails = SegmentStatusManager
+      .readTableStatusFile(CarbonTablePath.getTableStatusFilePath(table.getTablePath))
+    var segment0 = loadMetadataDetails.filter(x=> x.getLoadName.equalsIgnoreCase("0"))
+    Assert.assertEquals(692, segment0.head.getIndexSize.toLong)
```

Comment:
Do not assert for hard coded values. Instead create the file path and take its size using CarbonFile getSize interface method